gdnative-core/src/core_types/dictionary.rs

```diff
@@ -201,98 +230,86 @@ impl Dictionary<Shared> {
+    #[deprecated = "Care should be used when mutating shared variant collections. Prefer ThreadLocal or Unique collections unless you're absolutely sure that you want this."]
+    pub unsafe fn insert<K, V>(&self, key: K, val: V)
+    where
+        K: OwnedToVariant + ToVariantEq,
+        V: OwnedToVariant,
+    {
+        (get_api().godot_dictionary_set)(
+            self.sys_mut(),
+            key.owned_to_variant().sys(),
+            val.owned_to_variant().sys(),
+        )
     }
 
-    /// Returns an iterator through all key-value pairs in the `Dictionary`.
+    /// Erase a key-value pair in the `Dictionary` by the specified key.
     ///
-    /// `Dictionary` is reference-counted and have interior mutability in Rust parlance.
-    /// Modifying the same underlying collection while observing the safety assumptions will
-    /// not violate memory safely, but may lead to surprising behavior in the iterator.
+    /// # Safety
+    ///
+    /// It is only safe to call this on a shared dictionary when no other thread may
+    /// access the underlying collection during the call. Generally, it's not recommended to
+    /// mutate variant collections that may be shared. Prefer using `ThreadLocal` or `Unique`
+    /// collections unless you're absolutely sure that you want this.
     #[inline]
-    pub fn iter(&self) -> IterShared {
-        IterShared::new(self)
+    #[deprecated = "Care should be used when mutating shared variant collections. Prefer ThreadLocal or Unique collections unless you're absolutely sure that you want this."]
+    pub unsafe fn erase<K>(&self, key: K)
+    where
+        K: ToVariant + ToVariantEq,
+    {
+        (get_api().godot_dictionary_erase)(self.sys_mut(), key.to_variant().sys())
     }
 
-    /// Create a copy of the dictionary.
+    /// Clears the `Dictionary`, removing all key-value pairs.
     ///
-    /// This creates a new dictionary and is **not** a cheap reference count
-    /// increment.
+    /// # Safety
+    ///
+    /// It is only safe to call this on a shared dictionary when no other thread may
+    /// access the underlying collection during the call. Generally, it's not recommended to
+    /// mutate variant collections that may be shared. Prefer using `ThreadLocal` or `Unique`
+    /// collections unless you're absolutely sure that you want this.
     #[inline]
-    pub fn duplicate(&self) -> Dictionary<Unique> {
-        let d = Dictionary::new();
-        for (k, v) in self {
-            d.insert(&k, &v);
-        }
-        d
+    #[deprecated = "Care should be used when mutating shared variant collections. Prefer ThreadLocal or Unique collections unless you're absolutely sure that you want this."]
+    pub unsafe fn clear(&self) {
+        (get_api().godot_dictionary_clear)(self.sys_mut())
     }
 }
 
-/// Operations allowed on non-shared Dictionaries.
-impl Dictionary<Unique> {
-    /// Creates an empty `Dictionary`.
+impl Dictionary<ThreadLocal> {
```

Comment:
NIT doc comments on `impl` blocks do show up on doc pages which can be nice as a kind of overview when all items are folded.

Reply:
Added doc-comments on `impl` blocks.